category/category_repository_blackbox_test.go

```diff
@@ -0,0 +1,54 @@
+package category_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/almighty/almighty-core/category"
+	"github.com/almighty/almighty-core/gormsupport/cleaner"
+	"github.com/almighty/almighty-core/gormtestsupport"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type categoryRepoBlackBoxTest struct {
+	gormtestsupport.DBTestSuite
+	repo  category.Repository
+	clean func()
+	ctx   context.Context
+}
+
+func TestRunCategoryRepoBlackBoxTest(t *testing.T) {
+	suite.Run(t, &categoryRepoBlackBoxTest{DBTestSuite: gormtestsupport.NewDBTestSuite("../config.yaml")})
+}
+
+func (test *categoryRepoBlackBoxTest) SetupTest() {
+	test.repo = category.NewRepository(test.DB)
+	test.clean = cleaner.DeleteCreatedEntities(test.DB)
+}
+
+func (test *categoryRepoBlackBoxTest) TearDownTest() {
+	test.clean()
+}
+
+// TestCreateCategory tests that we can create a category
+func (test *categoryRepoBlackBoxTest) TestCreateLoadCategory() {
```

Comment:
Try to also test non-happy paths where you provoke a unique key violation and so forth. You can do this pretty elegantly by using named Go subtests and have one surrounding test method:

```go
// TestCreateCategory tests that we can create a category
func (test *categoryRepoBlackBoxTest) TestCreateLoadCategory() {

 	category := category.Category{
 		Name: "planner.testIssues",
 	}

 	test.T().Run("create and load (valid)", func(t *testing.T) {
 		category1, err := test.repo.Create(test.ctx, &category) // Create
 		require.Nil(test.T(), err)
 		require.NotNil(test.T(), category1)
 		require.NotNil(test.T(), category1.ID)
 		require.NotNil(test.T(), category1.Name)

 		category2, err := test.repo.LoadCategoryFromDB(test.ctx, category1.ID) // Load
 		require.Nil(test.T(), err)
 		require.NotNil(test.T(), category2)
 		require.NotNil(test.T(), category2.ID)
 
 		assert.Equal(test.T(), category1.ID, category2.ID)
 		assert.Equal(test.T(), category1.Name, category2.Name)
	}
	test.T().Run("create and load (invalid)", func(t *testing.T) {/* ... */}
	test.T().Run("create and load (unique name violation)", func(t *testing.T) {/* ... */}
	test.T().Run("create and load (not found cat)", func(t *testing.T) {/* ... */}
 }
```

Get the idea?